### PR TITLE
max_workers added 

### PR DESCRIPTION
**Pull request by NobodyIsPerfect78**
max_workers added as a parameter, this allows to set the number of workers when threading=true.

### CODE REVIEW

File: pynautobot/core/api.py [hunk 1]
```diff
@@ -74,6 +76,7 @@ def __init__(
         url,
         token=None,
         threading=False,
+        max_workers:int=4,
```
Comment (Kircheneer) [inline pynautobot/core/api.py:79]: I think given that we have type hints still in the docstring and no other type hints in the function signature, maybe we should drop these?

File: pynautobot/core/query.py [hunk 2]
```diff
@@ -131,6 +131,7 @@ def __init__(
         key=None,
         token=None,
         threading=False,
+        max_workers:int=4,
```
Comment (Kircheneer) [inline pynautobot/core/query.py:134]: Same here

**Review by Kircheneer — CHANGES_REQUESTED** [review]
Small nitpick, looks like a rather straight-forward change to me.

File: pynautobot/core/api.py (hunk 1 above)
Comment (NobodyIsPerfect78) [reply pynautobot/core/api.py]: fixed

File: pynautobot/core/query.py (hunk 2 above)
Comment (NobodyIsPerfect78) [reply pynautobot/core/query.py]: fixed

**Review by Kircheneer — APPROVED** [review]
LGTM

**Review by jvanderaa — APPROVED** [review]

**Pull request merged**